### ISSUE TITLE
CSV export delimiter setting

### ISSUE THREAD

**Issue by tofuSCHNITZEL** [netbox-community/netbox#17641]
### NetBox version

v4.0.11

### Feature type

Change to existing functionality

### Proposed functionality

There should be a global or user setting that defines the CSV delimiter used by the "Export CSV" function.

### Use case

Currently the CSV export is comma separated (as it should be!) but on every German Windows installation the default delimiter is the semicolon ";". so it would be helpful to have a setting to change the default separator of csv exports.

**Comment by pheus** [comment]
Hi everyone,

Thanks for the suggestion!

I've looked into the feasibility of supporting a customizable CSV delimiter. While the idea makes a lot of sense - especially for locales where `;` is the standard delimiter - it turns out that implementing this in the current NetBox setup isn't straightforward.

NetBox uses the `django_tables2.TableExport` class to handle CSV exports. Unfortunately, this class doesn’t currently support configurable delimiters. Enabling this feature would require overriding and maintaining a custom export method within the NetBox codebase.

For more context, here's a related discussion on the upstream issue:
[#899 Cannot set delimiter on CSV export](https://github.com/jieter/django-tables2/issues/899)

And here’s the relevant part of the NetBox code:
https://github.com/netbox-community/netbox/blob/ecb86567233e1ac4d4379590fdc7b6b45315767a/netbox/netbox/views/generic/bulk_views.py#L95-L99

Whether or not to take on this additional maintenance burden is ultimately up to the maintainers. That said, if support for configurable delimiters is added upstream in `django-tables2`, it would definitely simplify things and make this feature more viable in the future.

Thanks again for the thoughtful suggestion!

**Comment by jeremystretch** [comment]
Blocked by #19739

**Comment by pheus** [comment]
This request was later tracked in #19739 and has since been implemented in #19824, which shipped with **NetBox v4.4.0**. In v4.4.0+ you can set a per-user delimiter via **User Preferences** (user menu → Preferences → *CSV delimiter*).

To keep things tidy, I’m going to close this as a duplicate of #19739. Thanks again for taking the time to report it!
